packages/qdrant-loader/src/qdrant_loader/connectors/git/connector.py

```
@@ -432,6 +436,40 @@ async def get_documents(self) -> list[Document]:
+        return await super().get_documents()
+
+    async def fetch_by_id(self, entity_id: str) -> Document | None:
+        """Fetch a single file by its repo-relative path (forward slashes)."""
+        self._ensure_initialized()
+        file_path = resolve_safe_path(self.temp_dir, entity_id)
+        if file_path is None:
+            self.logger.warning(
+                "Path traversal attempt blocked", entity_id=entity_id
+            )
+            return None
+        if not os.path.exists(file_path) or not self.file_processor.should_process_file(  # type: ignore
```

Ruff (0.15.20)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)